### ISSUE TITLE
withScheme weird results

### ISSUE THREAD

**Issue by xobotyi** [thephpleague/uri#171]
### Bug Report

_(Fill in the relevant information below to help triage your issue.)_

| Information | Description |
|--------------|---------|
| Version | 6.2.1  |
| PHP version | 7.2 |
| OS Platform | Centos 6 |

#### Summary

`withScheme('')` should set make network-path reference, but it does invalid URI instead.

IMO when scheme set to empty string it shoud remove the scheme specification making the URI Network-Path Reference. And null to remove the scheme segment at all.

#### Standalone code, or other way to reproduce the problem

```php
Uri::createFromString('domain.com')->withScheme('');
```

#### Expected result

"//domain.com"

#### Actual result

":domain.com"

**Comment by nyamsprod** [comment]
@xobotyi you indeed did caught a bug (a regression from URI v5 😢 ) but it's not what you think 😄 

TL;DR:

```php
Uri::createFromString('domain.com')->withScheme('');
````
Should throw as an URI scheme can not be the empty string. Only PSR-7 allow the empty string but there's a specific class for that in the repo (ie the `Http` class).

Nevertheless the behaviour is expected because when you do

```php
$uri = Uri::createFromString('domain.com');
$uri->getHost(); // return null;
$uri->getPath(); //return 'domain.com'
````
you endup with an URI which only has a **path** set and not it's **host**. To set up the host you should do the following instead 

```php
$uri = Uri::createFromString('//domain.com');
$uri->getHost(); //return 'domain.com';
$uri->getPath(); //return ''
````

**Comment by xobotyi** [comment]
Huh. 
Anyways, as I understand there is no way to cast `https://domain.com` to `//domain.com`   without concatenation?

**Comment by nyamsprod** [comment]
yes there is

```php
<?php

$uri = Uri::createFromString('https://domain.com')->withScheme(null);

// or

$uri = Http::createFromString('https://domain.com')->withScheme('');
```

Depending on the class you are using
